### ISSUE TITLE
Set env variable pointing to home dir

### ISSUE THREAD

**Issue by nachokb** [shoes/shoes-deprecated#168]
When you package a Shoes app, the script gets embedded in a container which runs in the target platform (tried on Windows). It basically unpacks the script in a temp dir and runs it.

Inside the script however, Dir.pwd points to the temp dir. I couldn't find any way to get the real "current directory" which would be the directory the users executed the exe package (and similar in different platforms).

A simple way, I think, would be to update the runtime to set an env variables, somethings like ENV['SHOES_APP_DIR'] or Shoes.app_dir or something...

This is useful to access files which can be distributed besides the package (e.g. a conf file).

nachokb

**Comment by steveklabnik** [comment]
Hmm. On Windows? You should have a regular .exe. I thought. I was under the assumption that only Linux did such silliness...

**Comment by nachokb** [comment]
Yep it's windows. I don't understand what silliness would the Linux runtime be doing though :S

What I mean is: the exe resulting from my packaging an app, when it's ran, Dir.pwd points to C:\Windows\temp (or so). In that dir I find the .rb file that is executing. This makes sense actually (though I'd prefer it to create a subdir in the temp dir).

I think the files packaged together with the rb would also be unpacked there (couldn't try it: my packager hangs when packaging a directory, but that's a different issue).

So, I imagine the runtime (the SHOES_SETUP resource in the exe) is extracting the SHOES_PAYLOAD resource into the temp dir and running the Ruby script from there. Is there anyway to set an env variable before running the script? Couldn't find it in the source sadly...

Thanks for your time,

nachokb

**Comment by nachokb** [comment]
So, just an example:

I've got the following files in 'c:\users\foo\Desktop\myapp':
- myapp.exe
- somedata.dat

When the user clicks on myapp.exe (after installing the runtime if it's the first time), it unpacks SHOES_PAYLOAD in the temp dir and executes the Ruby script there (using the ruby vm installed the first time it runs).

At this moment, Dir.pwd == 'c:\windows\temp' and I need to get 'c:\users\foo\Desktop\myapp\somedata.dat'.

nachokb
